scripts/zones/Upper_Jeuno/npcs/Bheem.lua

```diff
@@ -1,23 +1,82 @@
 -----------------------------------
 -- Area: Upper Jeuno
---  NPC: Bheem
--- Standard Info NPC
+-- !pos: -90 0 169 244
+----------------------------------
+package.loaded["scripts/zones/Upper_Jeuno/TextIDs"] = nil
 -----------------------------------
-package.loaded["scripts/zones/Upper_Jeuno/TextIDs"] = nil;
------------------------------------
-require("scripts/zones/Upper_Jeuno/TextIDs");
+require("scripts/zones/Upper_Jeuno/TextIDs")
 -----------------------------------
 
 function onTrade(player,npc,trade)
-end;
+end
+
+function DECtoHEX(IN)
+    local B,K,OUT,I,D=16,"0123456789ABCDEF","",0
+    while IN>0 do
+        I=I+1
+        IN,D=math.floor(IN/B),math.mod(IN,B)+1
+        OUT=string.sub(K,D,D)..OUT
+    end
+    return OUT
+end
 
+function fellowParam(player)
```

Comment:
this needs to be done with bit ops instead of this - it is incredibly brittle and way more complex than it needs to be by assembling a number via a string and hoping the interpreter converts it right

Reply:
it is simply 4 bit.and (and 3 bit.lshifts) for the entirety of this function (and you can get rid of DECtoHEX)

Reply:
on that note, a giant `if/elseif` chain is usually a good sign to consider a table instead

Reply:
the table part is easy... but, for the bit shifts... i'd have to convert the values to binary first right? is there something i can look at already that uses bit ops on a bit packed param?